tests/backup_helper.go

```diff
@@ -10471,3 +10493,341 @@ func IsPxInstalled() bool {
+	}
+
+	if !AreStringSlicesEqual(groupIds, groups) {
+		return fmt.Errorf("Group list [%v] does not match the collaborators group list [%v] for cluster [%s]", groups, groupIds, clusterName)
+	}
+
+	// Expand the users list with members from groups
+	for _, group := range groups {
+		usersFromGroup, _ := backup.GetMembersOfGroupWithID(group)
+		for _, user := range usersFromGroup {
+			if !IsPresent(users, user) {
+				users = append(users, user)
+			}
+		}
+	}
+
+	// Validation that the cluster is accessible to shared users
+	errorChan := make(chan error, len(users))
+	defer close(errorChan)
+
+	var wg sync.WaitGroup
+	for _, user := range users {
+		wg.Add(1)
+		go func(user string) {
+			defer wg.Done()
+			nonAdminCtx, err := backup.GetNonAdminCtx(user, CommonPassword)
+			if err != nil {
+				errorChan <- err
+				return
+			}
+			log.Infof("Inspecting cluster [%s] with user [%s]", clusterName, user)
+			_, err = backupDriver.InspectCluster(nonAdminCtx, req)
+			if err != nil {
+				errorChan <- fmt.Errorf("User with id [%s] is not able to access the cluster [%s]", user, clusterName)
+			}
+		}(user)
+	}
+
+	wg.Wait()
+
+	// Collect errors from the channel
+	var errorMessages []string
+	for err := range errorChan {
+		errorMessages = append(errorMessages, err.Error())
+	}
+
+	if len(errorMessages) > 0 {
+		return fmt.Errorf("ValidateShareCluster Errors: %s", strings.Join(errorMessages, "; "))
+	}
+
+	return nil
+}
+
+// ValidateShareClusterBackup validates that a cluster's backups are shared with users and groups by validating the backup's access.
+func ValidateShareClusterBackup(ctx context1.Context, clusterName string, clusterUid string, users []string, groups []string) error {
+	backupDriver := Inst().Backup
+	clusterBackups, err := GetAllBackupsFromCluster(ctx, clusterName, clusterUid)
+	if err != nil {
+		return err
+	}
+
+	// Combine users from groups into the users slice
+	for _, group := range groups {
+		usersFromGroup, _ := backup.GetMembersOfGroupWithID(group)
+		users = append(users, usersFromGroup...)
+	}
+
+	errChan := make(chan error, len(clusterBackups)*len(users))
+
+	var wg sync.WaitGroup
+	for _, backupName := range clusterBackups {
+		for _, user := range users {
+			wg.Add(1)
+			go func(backupName, user string) {
+				defer wg.Done()
+				nonAdminCtx, err := backup.GetNonAdminCtx(user, CommonPassword)
+				if err != nil {
+					errChan <- err
+					return
+				}
+				backupInspectRequest := &api.BackupInspectRequest{
+					OrgId: BackupOrgID,
+					Name:  backupName,
+				}
+				log.Infof("Inspecting backup [%s] with user [%s]", backupName, user)
+				backupObj, err := backupDriver.InspectBackup(nonAdminCtx, backupInspectRequest)
+				if err != nil {
+					errChan <- fmt.Errorf("User with id [%s] is not able to access the backup [%s]", user, backupName)
+					return
+				}
+
+				if backupObj.GetBackup().UserBackupshareAccess != api.BackupShare_Restorable {
```

Comment:
These should be changed to create Restore from backup.

Reply:
same as above,  This serves as a basic validation to ensure the backup object has Restore access after the cluster is shared. All other CRUD operations will be validated at the test case level.